lib/puppet/feature/iis_compatible_version_installed.rb

```diff
@@ -0,0 +1,43 @@
+require 'puppet/util/feature'
+
+# Consider joining MajorVersion + MinorVersion instead of using VersionString
+
+def read_registry_for_iis_compatible_version
+  compatible_versions = ['7.5', '8.0', '8.5', '10.0']
+  installed_version = '0.0'
+
+  Puppet.debug "Reading Registry for IIS Version"
+
+  begin
+    require 'win32/registry'
+
+    f_HKLM        = Win32::Registry::HKEY_LOCAL_MACHINE
+    f_REG_PATH    = 'SOFTWARE\Microsoft\InetStp'
+    f_ACCESS_TYPE = Win32::Registry::KEY_READ | 0x100
```

Comment:
IIS 7.5 needs 2008R2+ ... Server SKUs from that point are only 64-bit. We don't allow 32-bit agents on 64-bit anymore.

Windows desktop SKUs can still be 32-bit, but we don't have to worry about reg redirection.

So I think this access is correct.

Reply:
The original PR reused code from lib/facter/iis_version.rb
I've rebased to include the changes in MODULES-5801 and MODULES-5954.